### PR TITLE
Implement various TypeScript conventions and patterns across the codebase

### LINKED ISSUE #98: Miscellaneous TypeScript pattern improvements
## Problem Statement
I needed a place to put miscellaneous changes to the directory structure and code base for making the TypeScript project for idiomatic.

## Proposed Solution
- 

## Alternative Proposals
- 

## Additional Context
- 

## Next Steps
- [ ] Team consensus to proceed 
- [ ] Complete Design Review Template (if applicable)
- [ ] Schedule Design Session

### PR DESCRIPTION
**Pull request by codefoster**
Everything is open for discussion.

## Type of PR

- [x] Documentation changes
  No relevant logging to update. Documentation could be added, but we'll do it as a separate task.

- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Issues Closed or Referenced

- Closes #98

### CODE REVIEW

File: src/app/controllers/ActorController.ts [hunk 1]
```diff
@@ -64,7 +62,7 @@ export class ActorController implements interfaces.Controller {
         
         if (!validated) {
             res.setHeader("Content-Type", "text/plain");
-            this.logger.Trace("InvalidParameter|" + "getAllActors" + "|" + message);
+            this.logger.trace("InvalidParameter|" + "getAllActors" + "|" + message);
```
Comment (bartr) [inline src/app/controllers/ActorController.ts]: In C#, we are tracing as a warning. We should standardize.

File: src/config/config.ts [hunk 2]
```diff
@@ -33,15 +33,15 @@ export async function getConfigValues(
...
+        log.error(Error(), "Failed to get required Cosmos DB secrets from KeyVault");
         process.exit(1);
     }
 
     // get optional App Insights Key
     try {
         insightsKey = await keyvault.getSecret(appInsightsKey);
     } catch {
-        log.Trace("Application Insights key not set.");
+        log.trace("Application Insights key not set.");
```
Comment (bartr) [inline src/config/config.ts]: I think this should be a warning so it appears in production logs at startup.

File: src/app/controllers/ActorController.ts (hunk 1 above)
Comment (aflinchb) [reply src/app/controllers/ActorController.ts]: Option to cover this in a separate PR with this issue: https://github.com/retaildevcrews/helium-typescript/issues/14

File: src/config/config.ts (hunk 2 above)
Comment (codefoster) [reply src/config/config.ts]: Agreed. I'll create an issue.

**Comment by codefoster** [comment]
@bartr, if I create an issue out of the comment that you raised, do you want to approve this PR so I can get it merged?

**Comment by codefoster** [comment]
Great. I'll use Issue #14

**Pull request merged**